plasmapy/particles/particle_collections.py

```diff
@@ -217,16 +211,24 @@ def data(self) -> List[Union[Particle, CustomParticle]]:
         A `list` containing the particles contained in the
         |ParticleList| instance.
 
-        The `~plasmapy.particles.particle_collections.ParticleList.data`
-        attribute should not be modified directly.
+        .. important::
```

Comment:
You already sort of enforce this by using a private attribute. You could do one better by adding a __setattr__ method for data that raises an exception?

Reply:
This one is kind of complicated.  `ParticleList` inherits `data` from `collections.UserList`, so we would need to keep `data` itself public to be consistent with the [Liskov Substitution Principle](https://reflectoring.io/lsp-explained/).  Even if we make `data` a `property`, it is still a `list` so we can do operations like:
```Python
>>> particle_list = ParticleList([])
>>> particle_list.data.append("not a particle")
```
The admonition is to warn against operations like that.

Reply:
Ok, got it. 

I think I've run into this problem before - would be nice if there were an abstract datatype like a list that isn't modifiable except by subclass methods. Like a 'secret list'